Expensify/App
# Fix waypoints removal

### Details

Fix the bug when it was impossible to remove a waypoint when editing a Distance request.

### Fixed Issues

$ https://github.com/Expensify/App/issues/34686
PROPOSAL: https://github.com/Expensify/App/issues/34686#issuecomment-1896980125


### Tests

Same as QA

### Offline tests

Same as QA

### QA Steps

1. Create a distance request with 3 waypoints
2. Navigate to the distance request details
3. Disable the internet connection
4. Click on the distance field to edit
5. Click on a waypoint
6. Click on the 3 dot menu > Delete waypoint
7. Hit Save
8. Go back online
- [ ] Verify the distance & amount are updated
- [ ] Verify the waypoint was deleted successfully

### PR Author Checklist



### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>


<details>
<summary>Android: mWeb Chrome</summary>

</details>


<details>
<summary>iOS: Native</summary>

</details>


<details>
<summary>iOS: mWeb Safari</summary>

</details>


<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/12595293/4cfe4ad5-93af-487c-8567-36f9d46b83cc


</details>


<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/12595293/8be92ad0-bf88-4a21-9fcf-d37fe1d5d78e


</details>
